src/averageAges.js

```diff
@@ -15,11 +15,21 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const filteredPeople = people.filter(person => {
+    const whatCentyry = Math.ceil(person.died / 100);
+
+    return century
+      ? person.sex === 'm'
+      && whatCentyry === century
+      : person.sex === 'm';
+  });
+  const averageAge = filteredPeople.reduce((acum, { born, died }) => {
+    const age = died - born;
+
+    return acum + age;
+  }, 0) / filteredPeople.length;
```

Comment:
Just move this to the separate function & reuse it
```suggestion
  const averageAge = getAverageAge(filteredPeople);
```